Update developer impacting changes (#2730)
Azure/azure-sdk: docs/policies/releasenotes.md

## What's a "developer impacting change"?

We don't want to advertise every single bug fix as most of them do not impact the way customers think about developing the client.  The [change log](https://azure.github.io/azure-sdk/policies_releases.html#change-logs) provides an exhaustive list of changes.  We don't need to duplicate it.

> However, in the release notes we do want to list critical changes for customers. A critical change is one that the developer would either need to know or want to know. Use the following section headers (*Features Added, Breaking Changes, and Key Bugs Fixed*) for the defined critical changes:

Reviewer: ```suggestion
However, in the release notes we do want to list critical changes for customers. A critical change is one that the developer would either need to know or want to know. Use the following section headers (`Features Added`, `Breaking Changes`, and `Key Bugs Fixed`) for the defined critical changes:
```

* *Features Added* - For new features to be called out in release notes
* *Breaking Changes* - For changes to be called out in release notes including, changes that break existing functionality, soon-to-be removed features, now removed features

> * *Key Bugs Fixed* - For important bug fixes to be called out in release notes,including any security fixes

Reviewer: ```suggestion
* *Key Bugs Fixed* - For important bug fixes to be called out in release notes, including any security fixes
```

> For example, "fixed a bug in which the event processor would stop working if you received no events for 30 minutes" does not qualify.  While you may list this under the "Fixed" heading in the changelog, the customer does not need to do anything, and it's fairly likely they have not bumped into this error, so this doesn't classify under "Key Bugs Fixed."  However, "Added a new overload to the constructor to support AzureAD credentials" would definitely be a good thing to include.

Reviewer: ```suggestion
For example, "fixed a bug in which the event processor would stop working if you received no events for 30 minutes" does not qualify.  While you may list this under the `Fixed` heading in the changelog, the customer does not need to do anything, and it's fairly likely they have not bumped into this error, so this shouldn't be listed under the `Key Bugs Fixed` heading.  However, "Added a new overload to the constructor to support AzureAD credentials" would definitely be a good thing to include.
```
